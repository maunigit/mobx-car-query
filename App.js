import React from "react";
import { View, SafeAreaView } from "react-native";
import { Provider } from "mobx-react";
import { DefaultTheme, DarkTheme, Provider as PaperProvider } from 'react-native-paper';
import Store from "./src/stores/Store";
import AppNavigator from './src/navigation/AppNavigator';
import globalStyles from './src/styles/Style';

/**
 * Navigable app
 */
export default class App extends React.Component {
  render() {
    return (
      //For use the store
      <Provider store={Store}>
        <PaperProvider theme={theme}>
          <SafeAreaView style={globalStyles.safeArea}>
            <View style={globalStyles.container}>
              <AppNavigator />
            </View>
          </SafeAreaView>
        </PaperProvider>
      </Provider>
    );
  }
}

/**
 * Theme of the app
 */
const theme = {
  ...DefaultTheme,
  dark: false, //must be false
  roundness: 8,
  colors: {
      ...DefaultTheme.colors,
      primary: 'blue',
      accent: 'aqua',
      error: 'red',
      text: 'black',   
      placeholder: 'orange',    
      notification: 'blueviolet',
      disabled: 'gray',
      background: 'darkolivegreen',
  }
};