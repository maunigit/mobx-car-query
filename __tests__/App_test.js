import React from 'react';
import App from '../App';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

it('App renders correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
