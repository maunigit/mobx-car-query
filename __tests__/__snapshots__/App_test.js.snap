// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Provider
  store={Store {}}
>
  <Provider
    theme={
      Object {
        "animation": Object {
          "scale": 1,
        },
        "colors": Object {
          "accent": "aqua",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "darkolivegreen",
          "disabled": "gray",
          "error": "red",
          "notification": "blueviolet",
          "onBackground": "#000000",
          "onSurface": "#000000",
          "placeholder": "orange",
          "primary": "blue",
          "surface": "#ffffff",
          "text": "black",
        },
        "dark": false,
        "fonts": Object {
          "light": Object {
            "fontFamily": "System",
            "fontWeight": "300",
          },
          "medium": Object {
            "fontFamily": "System",
            "fontWeight": "500",
          },
          "regular": Object {
            "fontFamily": "System",
            "fontWeight": "400",
          },
          "thin": Object {
            "fontFamily": "System",
            "fontWeight": "100",
          },
        },
        "roundness": 8,
      }
    }
  >
    <SafeAreaView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <NavigationContainer
          theme="light"
        />
      </View>
    </SafeAreaView>
  </Provider>
</Provider>
`;
